src/Toper/Request.php

```diff
@@ -128,7 +144,11 @@ public function send()
 
         while ($this->hostPool->hasNext()) {
             try {
-                $this->guzzleClient->setBaseUrl($this->hostPool->getNext());
+                if (!is_null($this->proxyDecorator)) {
+                    $this->guzzleClient->setBaseUrl($this->proxyDecorator->getBaseUrl());
```

Comment:
Ten if znajduje się w w while-u który odpala się wyłącznie wtedy gdy $this->hostPool->hasNext(). Czy tak powinno być? Czy w tej puli nie ma czasem usług pobranych z discovery? Co w przypadku gdy request ma iść przez proxy? Czy w tym przypadku w puli hostów coś w ogóle będzie?

Reply:
Poprawiłem, dziękuje za zwrócenie uwagi. W tej chwili w pooli coś jest, ale faktycznie w przyszłości może nie być

Reply:
Usunięte, faktycznie można wykorzystać StaticHostPoolProvider do ustawiania baseUrl w przypadku wykorzystania proxy